### PR TITLE
modify care team confirmation context for alerting

### PR DESCRIPTION
**Pull request by ewollesen**
Care team invitation/confirmation contexts are currently a clients.Permissions object. To support alerting, the context is extended to include a models.AlertsConfig.

The custom JSON unmarshaler allows flexibility in parsing the following received contexts:

    1. The existing permissions only context
    2. A hybrid context with old-style permissions plus an "AlertsConfig"
    3. A future context with a "Permissions" and an "AlertsConfig" as sibling
       properties

When the API is migrated to scenario #3 above, the custom marshaler can be removed.

Pulling in platform as a dependency brought a lot of vendor changes with it.

Part of BACK-2500

### CODE REVIEW

File: models/confirmation.go [hunk 1]
```diff
@@ -249,6 +251,64 @@ func generateKey() (string, error) {
 	}
 }
 
+// CareTeamContext specifies details associated with a Care Team Confirmation.
+type CareTeamContext struct {
```
Comment (ps2) [inline models/confirmation.go:255]: Does "CareTeam" refer to someone who is using the CarePartner app to care for someone looping, or is it more general than that?

Comment (ewollesen) [reply models/confirmation.go:255]: I believe it's more general. 

The terminology was already in the existing code. From my reading, it's any individual with permission to someone else's data. I don't see the term used for clinics/clinicians, only personal accounts. But for example, the existing invitation email, is loosely named "careteam invite" [Search: careteam](https://github.com/search?q=repo%3Atidepool-org%2Fhydrophone%20careteam&type=code)

Is it useful to make the distinction?

File: api/invite.go [hunk 2]
```diff
@@ -236,9 +236,13 @@ func (a *Api) AcceptInvite(res http.ResponseWriter, req *http.Request, vars map[
...
-		var permissions commonClients.Permissions
-		conf.DecodeContext(&permissions)
-		setPerms, err := a.gatekeeper.SetPermissions(inviteeID, invitorID, permissions)
+		ctc := &models.CareTeamContext{}
+		if err := conf.DecodeContext(ctc); err != nil {
+			statusErr := &status.StatusError{Status: status.NewStatus(http.StatusBadRequest, STATUS_ERR_DECODING_CONTEXT)}
+			a.sendModelAsResWithStatus(res, statusErr, http.StatusBadRequest)
+			return
+		}
+		setPerms, err := a.gatekeeper.SetPermissions(inviteeID, invitorID, ctc.Permissions)
```
Comment (toddkazakov) [inline api/invite.go:245]: I think permissions must include the new "follow" permission and validate that the new permission is present in the request body if alertsConfig is not empty.

Comment (ewollesen) [reply api/invite.go:245]: Also worth noting, that the new alerting permission isn't added yet. That happens in #126. I'm making a note to address it there.

**Review by toddkazakov — APPROVED** [review]

File: api/invite.go (hunk 2 above)
Comment (ewollesen) [reply api/invite.go:245]: I've now addressed this in #126.

**Comment by ewollesen** [comment]
Closing this, as it's been consumed by #126

**Pull request closed without merging**